Hazelcast.Test/Hazelcast.Client.Test/ClientSSLTest.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Hazelcast.Config;
+using Hazelcast.Core;
+using Hazelcast.Remote;
+using Hazelcast.Test;
+using NUnit.Framework;
+
+namespace Hazelcast.Client.Test
+{
+    [TestFixture]
+    public class ClientSSLTest : HazelcastTestSupport
+    {
+        private const string ValidCertName = "Hazelcast, Inc";
+
+        protected IHazelcastInstance Client { get; private set; }
+        protected RemoteController.Client RemoteController { get; private set; }
+
+
+        private ClientConfig ConfigureClient(bool ssl, bool validateCertificateChain, bool validateCertificateName, string certificateName)
+        {
+            var clientConfig = new ClientConfig();
+            clientConfig.GetNetworkConfig().AddAddress("localhost:5701");
+            clientConfig.GetNetworkConfig().GetSSLConfig().SetEnabled(ssl)
+                .SetProperty(SSLConfig.CertificateName, certificateName)
+                .SetProperty(SSLConfig.ValidateCertificateChain, validateCertificateChain.ToString())
+                .SetProperty(SSLConfig.ValidateCertificateName, validateCertificateName.ToString());
+            return clientConfig;
+        }
+
+
+        public void Setup(bool ssl, bool validateCertificateChain, bool validateCertificateName, string certificateName)
+        {
+            RemoteController = CreateRemoteController();
+            var cluster = CreateCluster(RemoteController, ssl?Resources.hazelcast_ssl:Resources.hazelcast);
+            RemoteController.startMember(cluster.Id);
+            Client = HazelcastClient.NewHazelcastClient(ConfigureClient(ssl, validateCertificateChain, validateCertificateName, certificateName));
+        }
+
+        [TearDown]
+        public void ShutdownRemoteController()
+        {
+            HazelcastClient.ShutdownAll();
+            StopRemoteController(RemoteController);
+        }
+
+        [Test]
+        public void TestMapSSLEnabled()
+        {
+            Setup(true, false, true, ValidCertName);
+            var map = Client.GetMap<string, string>(TestSupport.RandomString());
+            map.Put("key", "value");
```

Comment:
I think it would be clearer and enough to assert that client is connected, by checking lifecycle maybe?